detect_regressions.py

```diff
@@ -0,0 +1,184 @@
+import datetime
+import pathlib
+
+import pandas
+import sqlalchemy
+
+
+def detect_regressions(database_file):
+
+    engine = sqlalchemy.create_engine(f"sqlite:///{database_file}")
+
+    df = pandas.read_sql("select * from test_run where platform = 'linux'", engine)
+
+    # join runtime + py version
+    df = df.assign(
+        runtime=(
+            "coiled-"
+            + df.coiled_runtime_version
+            + "-py"
+            + df.python_version.str.split(".", n=2).str[:2].str.join(".")
+        ),
+        category=df.path.str.split("/", n=1).str[0],
+    )
+
+    # by_name_runtime = df.groupby(["name", "runtime"])
+    stats_dict = {}
+    regressions = []
+    reg_df = pandas.DataFrame(
+        columns=["category", "type", "mean", "last", "last-1", "last-2", "threshold"]
+    )
+
+    runtimes = list(df.runtime.unique())
+    for runtime in runtimes:
+        by_test = df[(df.runtime == runtime)].groupby("name")
+
+        test_names = list(by_test.groups.keys())
+        for name in test_names:
+            df_test = by_test.get_group(name)
+
+            df_passed = df_test[df_test.call_outcome == "passed"]
+            # check for empty dataframe and having enough data to do some stats
+            # currently we don't have a lot of data but eventually use 10 instead of 6.
+
+            if not df_passed.empty and (len(df_passed) >= 6):
+
+                # check the test is not obsolete.
+                date_last_run = datetime.datetime.strptime(
+                    df_passed.start.values[-1].split()[0], "%Y-%m-%d"
+                )
+                date_threshold = datetime.datetime.today() - datetime.timedelta(7)
+
+                if date_last_run < date_threshold:
+                    # the latest run was 7+ days ago, test is obsolete
+                    pass
```

Comment:
We could probably work this into the SQL query as well, though it would take a bit more work. In the meantime, I think it would be a bit cleaner to use the pandas util:
```python
if pd.to_timestamp(df_passed.start.iloc[-1]) < pd.Timestamp.now() - pd.Timedelta(days=7):
    pass
```

(I'm not sure why pandas is having a tough time with the datetime column in the db)